### PR TITLE
FDP-1628: added remote psk store

### PR DESCRIPTION
**Pull request by LucianoFavoroso**

### CODE REVIEW

**Review by smvdheijden — CHANGES_REQUESTED** [review]

File: application/src/main/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/coap/configuration/psk/RemotePskStore.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package org.gxf.standalonenotifyinggateway.coaphttpproxy.coap.configuration.psk
```
Comment (smvdheijden) [inline application/src/main/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/coap/configuration/psk/RemotePskStore.kt:1]: Missing license header

**Review by smvdheijden — CHANGES_REQUESTED** [review]

File: application/src/main/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/coap/configuration/CoapConfiguration.kt [hunk 2]
```diff
@@ -37,11 +38,8 @@ class CoapConfiguration(private val coapProps: CoapProperties, private val udpPr
     }
 
     @Bean
-    fun pskStore(pskStoreStub: PskStoreStub) =
-            AdvancedMultiPskStore()
-                    .apply {
-                        pskStoreStub.retrieveAll().forEach { psk -> this.setKey(psk.id, psk.key.toByteArray()) }
-                    }
+    fun pskStore(webClient: WebClient, remoteLogger: RemoteLogger) =
+            RemotePskStore(webClient, remoteLogger)
```
[the remark this answers is not in the document]
Comment (LucianoFavoroso) [reply application/src/main/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/coap/configuration/CoapConfiguration.kt:42]: Removed bean, it is not needed for correct initialization.

File: application/src/main/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/coap/configuration/psk/RemotePskStore.kt (hunk 1 above)
Comment (LucianoFavoroso) [reply application/src/main/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/coap/configuration/psk/RemotePskStore.kt:5]: Added header but github doesn't show outdated

**Review by JelleHoffman — APPROVED** [review]

File: application/src/integrationTest/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/IntegrationTest.kt [hunk 3]
```diff
@@ -36,17 +35,23 @@ class IntegrationTest {
     @Value("\${config.psk.default-id}")
```
Comment (sanderv) [inline application/src/integrationTest/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/IntegrationTest.kt:35]: Waarom gebruik je hier niet gewoon `PskStubProperties`?

File: application/src/test/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/MessageHandlerTest.kt [hunk 4]
```diff
@@ -43,25 +45,24 @@ class MessageHandlerTest {
...
-        Mockito.`when`(messageValidator.isValid(any<Message>())).thenReturn(true)
+        `when`(messageValidator.isValid(any<Message>())).thenReturn(true)
 
         messageHandler.handlePost("12345", testCbor)
 
-        Mockito.verify(httpClient).postMessage(check {
-            Assertions.assertEquals(message.deviceId, it.deviceId)
-            Assertions.assertEquals(message.payload, it.payload)
+        verify(httpClient).postMessage(check {
+            assertThat(message).usingRecursiveComparison().isEqualTo(it)
```
Comment (smvdheijden) [inline application/src/test/kotlin/org/gxf/standalonenotifyinggateway/coaphttpproxy/MessageHandlerTest.kt:65]: Wrong order of actual object under test and the expected value.
```suggestion
            assertThat(it).usingRecursiveComparison().isEqualTo(message)
```

File: application/build.gradle.kts [hunk 5]
```diff
@@ -23,6 +23,7 @@ dependencies {
     testImplementation("org.junit.jupiter:junit-jupiter")
     testImplementation("org.mockito:mockito-junit-jupiter")
     testImplementation("org.mockito.kotlin:mockito-kotlin:5.1.0")
+    testImplementation("org.assertj:assertj-core:3.24.2")
```
Comment (smvdheijden) [inline application/build.gradle.kts:26]: Isn't the version provided by Spring Boot?

**Review by smvdheijden — APPROVED** [review]

**Pull request merged**